pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -136,6 +139,52 @@ func (m *managerImpl) IsUnderDiskPressure() bool {
 	return hasNodeCondition(m.nodeConditions, api.NodeDiskPressure)
 }
 
+func (m *managerImpl) createThresholdNotification(diskInfoProvider DiskInfoProvider, podFunc ActivePodsFunc, observations signalObservations) error {
+	// default to no-op notifier
+	m.thresholdNotifier = NewNoOpThresholdNotifier()
+
+	thresholds := m.config.Thresholds
+	if len(thresholds) == 0 {
+		return nil
+	}
+
+	for _, threshold := range thresholds {
+		// only enable memcg threshold notification if a hard memory eviction limit is set
+		if threshold.Signal != SignalMemoryAvailable || threshold.GracePeriod != time.Duration(0) {
```

Comment:
is the complexity that we need to have multiple notifiers?  i think for a first pass, just having it for eviction-hard is fine, but in the long run, i would want both.  ideally, we could have the memcg notification that we crossed the threshold, and set a timer to wake up at the associated grace period and run sync at that point.